tests/test_cases/socketselect.c

```diff
@@ -0,0 +1,331 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <errno.h>
+#include <string.h>
+#include <fcntl.h> 
+#include <sys/select.h>
+#include <sys/types.h>
+#include <sys/ioctl.h>
+#include <sys/socket.h>
+#include <sys/time.h>
+#include <netinet/in.h>
+#include <arpa/inet.h> 
+#include <pthread.h> 
+
+#define PORT 9995
+
+#define TRUE             1
+#define FALSE            0
+
+void* client(void* v) { 
+    int sock = 0, valread; 
+    struct sockaddr_in serv_addr; 
+    char *hello = "Hello from client"; 
+    char buffer[1024] = {0}; 
+    sock = socket(AF_INET, SOCK_STREAM, 0);
+    serv_addr.sin_family = AF_INET; 
+    serv_addr.sin_port = htons(PORT); 
+       
+    // Convert IPv4 and IPv6 addresses from text to binary form 
+    inet_pton(AF_INET, "127.0.0.1", &serv_addr.sin_addr);
+
+    sleep(1);
+   
+    connect(sock, (struct sockaddr *)&serv_addr, sizeof(serv_addr));
+    send(sock , hello , strlen(hello) , 0 ); 
+    printf("Hello message sent\n"); 
+    valread = read( sock , buffer, 1024); 
+    printf("%s\n",buffer ); 
+    fflush(stdout);
+    close(sock);
+    return NULL; 
+} 
+
+void* server(void* v) {
+   int    i, len, rc, on = 1;
+   int    listen_sd, max_sd, new_sd;
+   int    desc_ready, end_server = FALSE;
+   int    close_conn;
+   char   buffer[80];
+   struct sockaddr_in   addr;
+   struct timeval       timeout;
+   fd_set        master_set, working_set;
+
+   
+   /* Create an AF_INET6 stream socket to receive incoming      */
+   /* connections on                                            */
+   
+   listen_sd = socket(AF_INET, SOCK_STREAM, 0);
+   if (listen_sd < 0)
+   {
+      perror("socket() failed");
+      exit(-1);
+   }
+   
+   /* Allow socket descriptor to be reuseable                   */
+   
+   rc = setsockopt(listen_sd, SOL_SOCKET,  SO_REUSEADDR,
+                   (char *)&on, sizeof(on));
+   if (rc < 0)
+   {
+      perror("setsockopt() failed");
+      close(listen_sd);
+      exit(-1);
+   }
+
+   
+   /* Set socket to be nonblocking. All of the sockets for      */
+   /* the incoming connections will also be nonblocking since   */
+   /* they will inherit that state from the listening socket.   */
+   
+   rc = ioctl(listen_sd, FIONBIO, (char *)&on);
+   if (rc < 0)
+   {
+      perror("ioctl() failed");
+      close(listen_sd);
+      exit(-1);
+   }
+
+   
+   /* Bind the socket                                           */
+   
+   memset(&addr, 0, sizeof(addr));
```

Comment:
No need to memset if you will just memcpy over that

Reply:
This was from some of the code I ported, fine with it being removed.